### ISSUE TITLE
csurf package deprecated; find replacement / phase it out

### ISSUE THREAD

**Issue by dlongley** [digitalbazaar/bedrock-views#67]
See: https://www.npmjs.com/package/csurf

**Comment by dlongley** [comment]
One option here is to remove the csurf package and, if possible, just disallow URL-encoded bodies on any route by default. If an application needs special route that uses URL-encoded bodies (like for old-style OAuth2 flows or something like that), it can enable it and require some kind of CSRF token using whatever package is appropriate.

**Comment by dlongley** [comment]
I'm marking this as P2 so we can get it removed and clean up any vulnerability trackers that are flagging applications that use `csurf`. It's not P1 because none of our modern applications have routes that accept URL-encoded request bodies -- and we recommend against that practice generally because it's harder to defend against attacks.

**Comment by mattcollier** [comment]
@dlongley we need to get this assigned to an engineer soon.  There is a vulnerability in `cookie@0.4.0` that has been addressed in v0.7.0.

```
│ ├─┬ csurf@1.11.0
│ │ ├── cookie-signature@1.0.6 deduped
│ │ ├── cookie@0.4.0
│ │ ├─┬ csrf@3.1.0
│ │ │ ├── rndm@1.2.0
│ │ │ ├── tsscmp@1.0.6
│ │ │ └── uid-safe@2.1.5 deduped
```

**Comment by dlongley** [comment]
The top candidate replacement package is: https://www.npmjs.com/package/csrf-csrf

Someone will need to explore integrating and testing it with some apps.

**Comment by dlongley** [comment]
Notably, my preference is still to remove this entirely and require any route that needs CSRF protection (i.e., because it's cookie-authenticated *and* accepting POSTs using `application/x-www-form-urlencoded` content) add it in themselves as appropriate. This, of course, would be a breaking change.

**Comment by dlongley** [comment]
Addressed by #73.
